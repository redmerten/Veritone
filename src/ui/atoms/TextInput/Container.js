// @flow

import { rem } from 'polished';
import styled, { css } from 'styled-components';

export default styled.div`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.antiFlashWhite};
  border: ${({ theme }) => theme.colors.antiFlashWhite} solid ${rem(2)};
  border-radius: ${rem(4)};
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'initial')};
  display: flex;
  min-height: ${rem(38)};
  padding: 0 ${({ theme }) => rem(theme.spacing.md)};
  transition: all 0.25s ${({ theme }) => theme.timing.easeOutCirc};
  width: 100%;

  &:hover {
    background-color: ${({ theme }) => theme.colors.azureishWhite};
    border-color: ${({ theme }) => theme.colors.azureishWhite};
  }

  &:focus-within {
    background-color: ${({ theme }) => theme.colors.white};
    border-color: ${({ theme }) => theme.colors.gainsboro};
  }

  & input:-webkit-autofill {
    box-shadow: inset 0 0 0px 9999px ${({ theme }) => theme.colors.antiFlashWhite};

    &:hover {
      box-shadow: inset 0 0 0px 9999px ${({ theme }) => theme.colors.azureishWhite};
    }

    &:focus {
      box-shadow: inset 0 0 0px 9999px ${({ theme }) => theme.colors.white};
    }
  }

  ${({ type }) =>
    type &&
    type === 'number' &&
    css`
      padding-right: ${({ theme }) => rem(theme.spacing.sm)};
    `};

  ${({ small, theme }) =>
    small &&
    css`
      display: inline-flex;
      height: ${rem(32)};
      padding: 0 ${rem(theme.spacing.sm)};
      width: auto !important;

      /* stylelint-disable-next-line no-descending-specificity */
      input {
        width: ${rem(32)};
      }
    `};
`;
