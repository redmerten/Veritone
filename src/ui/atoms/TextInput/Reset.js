// @flow

import { rem } from 'polished';
import styled, { css } from 'styled-components';

import { ReactComponent } from './reset.svg';

export default styled(ReactComponent)`
  cursor: pointer;
  height: ${rem(24)};
  width: ${rem(24)};

  path {
    fill: ${({ theme }) => theme.colors.ashGrey};
  }

  /* stylelint-disable no-descending-specificity */
  &:hover path {
    fill: ${({ theme }) => theme.colors.grayBlue};
  }

  ${({ disabled, theme }) =>
    disabled &&
    css`
      cursor: default;
      path {
        fill: ${theme.colors.gainsboro} !important;
      }
    `};
`;
