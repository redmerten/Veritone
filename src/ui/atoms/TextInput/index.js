// @flow

import React from 'react';

import Input from '../Input';

import Container from './Container';
import Reset from './Reset';

type Props = {
  dataTestId?: string,
  disabled?: boolean,
  onFocus?: () => void,
  onReset?: () => void,
  placeholder?: string,
  resetable?: boolean,
  small?: boolean,
  type?: 'password' | 'number' | 'text',
};

function TextInput(
  {
    dataTestId = 'atom-text-input',
    disabled = false,
    onReset,
    resetable = false,
    small = false,
    type = 'text',
    placeholder = '',
    ...rest
  }: Props,
  ref
) {
  return (
    <Container data-testid={dataTestId} disabled={disabled} small={small} type={type}>
      <Input
        disabled={disabled}
        placeholder={placeholder}
        ref={ref}
        tabIndex={disabled ? -1 : 0}
        type={type}
        {...rest}
      />
      {resetable && <Reset disabled={disabled} onClick={!disabled ? onReset : undefined} />}
    </Container>
  );
}

// $FlowIssue
export default React.forwardRef(TextInput);
