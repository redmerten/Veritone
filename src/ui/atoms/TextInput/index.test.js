// @flow

import React from 'react';

import TextInput from '.';

import { render } from 'utils/testing';

describe('atoms/TextInput', () => {
  it('matches its base snapshot', () => {
    const { baseElement } = render(<TextInput />);
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('renders TextInput atom', () => {
    const { getByTestId } = render(<TextInput />);
    expect(getByTestId('atom-text-input')).toBeInTheDocument();
  });
});
