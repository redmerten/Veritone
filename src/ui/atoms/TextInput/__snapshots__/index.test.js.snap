// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/TextInput matches its base snapshot 1`] = `
.c1 {
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  appearance: textfield;
  background: none;
  border: 0;
  color: #242425;
  cursor: initial;
  font-family: 'roboto-mono',monospace;
  font-size: 0.75rem;
  font-weight: 400;
  margin: 0;
  outline: 0;
  padding: 0;
  width: inherit;
}

.c1:-webkit-autofill {
  box-shadow: inset 0 0 0px 9999px #f2f3f5;
}

.c1::-webkit-input-placeholder {
  color: #8F95a5;
}

.c1::-moz-placeholder {
  color: #8F95a5;
}

.c1:-ms-input-placeholder {
  color: #8F95a5;
}

.c1::placeholder {
  color: #8F95a5;
}

.c1:disabled {
  color: #B0BEC5 !important;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f2f3f5;
  border: #f2f3f5 solid 0.125rem;
  border-radius: 0.25rem;
  cursor: initial;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 2.375rem;
  padding: 0 1rem;
  -webkit-transition: all 0.25s cubic-bezier(0.075,0.82,0.165,1);
  transition: all 0.25s cubic-bezier(0.075,0.82,0.165,1);
  width: 100%;
}

.c0:hover {
  background-color: #e1e7ee;
  border-color: #e1e7ee;
}

.c0:focus-within {
  background-color: #fff;
  border-color: #d8e0ea;
}

.c0 input:-webkit-autofill {
  box-shadow: inset 0 0 0px 9999px #f2f3f5;
}

.c0 input:-webkit-autofill:hover {
  box-shadow: inset 0 0 0px 9999px #e1e7ee;
}

.c0 input:-webkit-autofill:focus {
  box-shadow: inset 0 0 0px 9999px #fff;
}

<div>
  <div
    class="c0"
    data-testid="atom-text-input"
    type="text"
  >
    <input
      class="c1"
      data-testid="atom-input"
      placeholder=""
      tabindex="0"
      type="text"
    />
  </div>
</div>
`;
